// porn-x/src/videos/getVideos.js
import got from "got";
import cheerio from "cheerio";
import ora from "ora";

/**
 * @param {String} query
 * @param {{ page?: number}} options
 */
export default async function retrieveData(query, options = {}) {
  const spinner = ora("Started fetching...").start();
  const request = await got(
    `https://www.xtits.com/search/${query}/?mode=async&function=get_block&block_id=list_videos_videos_list_search_result&q=${query}&category_ids=&sort_by=&from_videos=${
      !options?.page ? 1 : options?.page
    }`
  );
  const body = request.body;
  const $ = cheerio.load(body);
  const array = [];
  const videoList = [];
  $(".thumbs-holder")
    .find(".item")
    .map(async (index, element) => {
      const $element = $(element);
      const video = $element.find("a").last().attr("href");
      videoList.push(video);
    });
  for (const url of videoList) {
    spinner.color = "yellow";
    spinner.text = `Fetched ${array.length} videos`;
    const data = await getVideo(url);
    array.push({
      uri: data,
    });
  }

  spinner.stopAndPersist({
    symbol: "✅",
    text: `Successfully fetched ${array.length} videos`,
    prefixText: "",
    suffixText: "",
  });

  return array;
}
async function getVideo(url) {
  const request = await got(url);
  const response = request.body;
  const $ = cheerio.load(response);
  const data = $(".content-holder")
    .find(".player-holder")
    .find("script")
    .last()
    .html();
  const jsonData = data.substring(data.indexOf("flashvars"));
  var regex = /video_url: '([^']+)'/;
  var match = jsonData.match(regex);
  var videoUrl = match[1];
  return videoUrl;
}
