// porn-x/src/functions/getStarScenes.js
import got from "got";
import cheerio from "cheerio";
import google from "googlethis";

export async function getStarScenes(query) {
  const $queryToSearch = `adultfilmdatabase ${query} actor`;
  const $this = await google.search($queryToSearch, {
    safe: false,
    parse_ads: false,
  });
  const $filteredArray = $this.results.filter((result) =>
    result.url.includes("adultfilmdatabase")
  );
  const $urlToFetch = $filteredArray[0].url;
  const baseURL = `${$urlToFetch}`;
  const request = await got(baseURL);
  const body = request.body;
  const $ = cheerio.load(body);
  const array = [];
  const name = $("h1.w3-opacity").text().trim();
  $(`tbody`)
    .find("tr")
    .map((_, el) => {
      const $element = $(el);
      const name = $element.find("a").text();
      const year = $element.find("td:nth-child(4)").text();
      const genres = $element.find("td:nth-child(3)").text().split(",");
      const studio = $element.find("td:nth-child(2)").text();
      const href = $element.find("a").attr("href");
      const thumbnailPath = $element.find("a").attr("actorthumb");
      const thumbnail = `https://www.adultfilmdatabase.com${thumbnailPath}`;
      array.push({
        title: name,
        studio,
        genres,
        year,
        href,
        thumbnail,
      });
    });
  const object = {
    name,
    total: array.length,
    scenes: array,
  };
  return object;
}
